### ISSUE TITLE
[Bug]: Parsing Nested Font Styles Does Not Clear Marks

### ISSUE THREAD

**Issue by mgreystone** [ueberdosis/tiptap#5585]
### Affected Packages

core

### Version(s)

2.6.6

### Bug Description

When using marks like bold or italic, when parsing `font` CSS rules, nested rules do not clear a parent mark. This effects how content is parsed, either from the clipboard or directly passed to the tiptap editor.

For example, if we have HTML with nested `font-weight` rules like so & parse with the `StarterKit`...

```html
<p style="font-weight: 700;">
  <span style="font-weight: 700;">I am Bold</span>
  <span style="font-weight: 400;">I am not Bold</span>
</p>
```

We get a document that looks like this.
```
doc(paragraph(strong("I am Bold "), "I am not Bold"))
```

Instead, we get a document that looks like this.
```
doc(paragraph(bold("I am Bold I am not Bold")))
```

---

We can workaround this behavior by extending the `parseHTML` method of the `Bold` extension with a `clearMark` directive.

```
Bold.extend({
  parseHTML() {
    return [
      {
        tag: "strong",
      },
      {
        tag: "b",
        getAttrs: (node) => node.style.fontWeight != "normal" && null,
      },
      {
        style: "font-weight=400",
        clearMark: (mark) => mark.type.name === this.name,
      },
      {
        style: "font-weight",
        getAttrs: (value) => /^(bold(er)?|[5-9]\d{2,})$/.test(value) && null,
      },
    ];
  },
}),
```

---

For what it's worth, the `prosemirror-schema-basic` package [includes this behavior](https://github.com/ProseMirror/prosemirror-schema-basic/blob/4e358f5443f15abe09406c9327e35aa9e3c1bf8c/src/schema-basic.ts#L145) and clears out parent marks & works how i would expect.

### Browser Used

Chrome

### Code Example URL

https://codesandbox.io/p/sandbox/gracious-darkness-nx3s7s?file=%2Fsrc%2Findex.js

### Expected Behavior

I would expect the mark parsers to clear out marks when appropriate & roughly match the input HTML.

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by nperez0111** [comment]
Feel free to contribute a PR with this fix

**Comment by nperez0111** [comment]
This is released with v2.9.0
